src/Client/Web/Bit.Client.Web.BlazorUI/Components/SpinButtons/BitSpinButton.razor.cs

```diff
@@ -0,0 +1,147 @@
+﻿using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Web;
+using Microsoft.JSInterop;
+using System;
+using System.Threading.Tasks;
+
+namespace Bit.Client.Web.BlazorUI
+{
+    public partial class BitSpinButton
+    {
+        private LabelPosition labelPosition = LabelPosition.left;
+        public ElementReference InputElement { get; set; }
+        public double Value { get; set; }
+
+        [Inject] public IJSRuntime JSRuntime { get; set; }
+        
+        [Parameter] public double Min { get; set; } = 0;
+        [Parameter] public double Max { get; set; } = double.MaxValue;
+        [Parameter] public double DefaultValue { get; set; } = 0;
+        [Parameter] public double Step { get; set; } = 1;
+        [Parameter] public string Suffix { get; set; }
+        [Parameter] public string Label { get; set; }
+        [Parameter] public string IconName { get; set; }
+        [Parameter] public LabelPosition LabelPosition { 
+            get => labelPosition; 
+            set {
+                labelPosition = value;
+                ClassBuilder.Reset();
+            }}
+        [Parameter] public EventCallback<string> OnChange { get; set; }
+
+        protected virtual async Task HandleUpClick(MouseEventArgs e)
+        {
+            if (IsEnabled)
+            {
+                var result = Value + Step;
+                if (result <= Max)
+                {
+                    Value = Normalize(result);
+                    await OnChange.InvokeAsync(ValueWithSuffix);
+                }
+            }
+        }
+
+        protected virtual async Task HandleDownClick(MouseEventArgs e)
+        {
+            if (IsEnabled)
+            {
+                var result = Value - Step;
+                if (result >= Min)
+                {
+                    Value = Normalize(result);
+                    await OnChange.InvokeAsync(ValueWithSuffix);
+                }
+            }
+        }
+
+        protected override string RootElementClass => "bit-spin-button";
+
+        protected override void RegisterComponentClasses()
+        {
+            ClassBuilder.Register(() => $"label-{LabelPosition}");
```

Comment:
This implementation is kinda not correct.
Please take a look at other components (like BitButton) to grasp the correct implementation of this method.